File: examples/basic-agent/wait.mjs
#!/usr/bin/env node
// Poll a card until verdict, then close the loop if approved.
// Usage: node wait.mjs <card_id>

import { Client } from '../../sdk/node/index.mjs';

const cardId = process.argv[2];
if (!cardId) {
  console.error('usage: wait.mjs <card_id>');
  process.exit(2);
}

const DESK = process.env.DECISION_DESK_URL || 'http://127.0.0.1:3335';
const dd = new Client(DESK, 'basic-agent');

console.log(`polling ${cardId} for verdict (Ctrl-C to stop)...`);
const verdict = await dd.waitForDecision(cardId, { timeoutMs: 3600_000, pollMs: 3000 });
console.log(`verdict: ${verdict.status}`);
const last = verdict.decisions?.[0];
if (last?.notes) console.log(`  notes: ${last.notes}`);
if (last?.denial_tags) console.log(`  denial tags: ${last.denial_tags}`);

if (verdict.status === 'approved') {
  console.log('doing the work... (pretend)');
  const result = await dd.markExecuted(cardId, {
    execution_status: 'completed',
    execution_notes: 'node example finished',
  });
  console.log(`executed at ${result.executed_at}`);
} else if (verdict.status === 'denied') {
  console.log('noted. Agent should learn from the denial tags above.');
  console.log('(see GET /api/intelligence for the running pattern list.)');
}

File: examples/basic-agent/wait.py
#!/usr/bin/env python3
"""Poll a card until the reviewer approves or denies it.

Usage: python wait.py <card_id>

If approved, this script also marks the card executed — closing the loop.
"""

import os
import sys

sys.path.insert(0, os.path.join(os.path.dirname(__file__), "..", "..", "sdk", "python"))
from decision_desk import Client  # noqa: E402

if len(sys.argv) < 2:
    print("usage: wait.py <card_id>", file=sys.stderr)
    sys.exit(2)

card_id = sys.argv[1]
DESK = os.environ.get("DECISION_DESK_URL", "http://127.0.0.1:3335")

dd = Client(DESK, agent_id="basic-agent")
print(f"polling {card_id} for verdict (Ctrl-C to stop)...")
verdict = dd.wait_for_decision(card_id, timeout=3600, poll_interval=3)
print(f"verdict: {verdict['status']}")
if verdict.get("decisions"):
    last = verdict["decisions"][0]
    if last.get("notes"):
        print(f"  notes: {last['notes']}")
    if last.get("denial_tags"):
        print(f"  denial tags: {last['denial_tags']}")

if verdict["status"] == "approved":
    print("doing the work... (pretend)")
    result = dd.mark_executed(
        card_id,
        execution_status="completed",
        execution_notes="example agent finished — sent via SMTP",
    )
    print(f"executed at {result['executed_at']}")
elif verdict["status"] == "denied":
    print("noted. The agent should learn from the denial tags above.")
    print("(see GET /api/intelligence for the running pattern list.)")
